doc/arch/system_context.puml

```diff
@@ -0,0 +1,33 @@
+@startuml
+!include https://raw.githubusercontent.com/plantuml-stdlib/C4-PlantUML/master/C4_Context.puml
+!include https://raw.githubusercontent.com/plantuml-stdlib/C4-PlantUML/master/C4_Container.puml
+' uncomment the following line and comment the first to use locally
+' !include C4_Context.puml
+
+LAYOUT_TOP_DOWN()
+
+title System-level component diagram for PennyLane using Lightning Qubit device
+
+Person(developer, "Quantum software developer")
+System(python, "Python", "Python 3.7+")
```

Comment:
Now I know that these are autogenerated from an easily-searcheable script, maybe putting Python 3.7 isn't so bad here!